lms/views/api/blackboard/files.py

```diff
@@ -66,11 +67,30 @@ def via_url(self):
         """Return the Via URL for annotating the given Blackboard file."""
 
         course_id = self.request.matchdict["course_id"]
+        course = self.request.find_service(name="course").get_by_context_id(course_id)
```

Comment:
Now on the view use the plugin to handle course copy.

This code will be very similar in BB/Canvas and D2L, we  can find a common abstraction later. Maybe even a common `list_files` view for all of them.

I rather tackle that after the course copy issues are solved.